registry/remote/example_test.go

```diff
@@ -0,0 +1,624 @@
+		panic(err)
+	}
+
+	fmt.Println(string(pulledBlob))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_FetchReference_manifestByDigest gives example snippets for downloading a manifest by digest.
+func ExampleRepository_FetchReference_manifestByDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	descriptor, rc, err := repo.FetchReference(ctx, digest)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+	pulled, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulled) || descriptor.Size != int64(len(pulled)) {
+		panic(err)
+	}
+
+	fmt.Println(string(pulled))
+
+	// Output:
+	// Example manifest content
+}
+
+// ExampleRepository_Fetch_layer gives example snippets for downloading a layer blob by digest.
+func ExampleRepository_Fetch_layer() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	descriptor, err := repo.Blobs().Resolve(ctx, digest.FromBytes([]byte(exampleLayer)).String())
+	if err != nil {
+		panic(err)
+	}
+	rc, err := repo.Fetch(ctx, descriptor)
+	if err != nil {
+		panic(err)
+	}
+	defer rc.Close() // don't forget to close
+
+	// option 1: fetch a WHOLE blob
+	pulledBlob, err := io.ReadAll(rc)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(string(pulledBlob))
+
+	// verify the fetched content
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err)
+	}
+
+	// option 2: partially read, if the remote registry supports
+	if seeker, ok := rc.(io.ReadSeeker); ok {
+		offset := int64(8)
+		_, err = seeker.Seek(offset, io.SeekStart)
+		if err != nil {
+			panic(err)
+		}
+		pulledBlob, err := io.ReadAll(rc)
+		if err != nil {
+			panic(err)
+		}
+		if descriptor.Size-offset != int64(len(pulledBlob)) {
+			panic(err)
+		}
+		fmt.Println(string(pulledBlob))
+	}
+
+	// Output:
+	// Example layer content
+	// layer content
+}
+
+// ExampleRepository_Tag gives example snippets for tagging a manifest.
+func ExampleRepository_Tag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	// tag a manifest referenced by the digest below
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	descriptor, err := repo.Resolve(ctx, digest)
+	if err != nil {
+		panic(err)
+	}
+	tag := "latest"
+	err = repo.Tag(ctx, descriptor, tag)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
@Wwwsylvia `Resolve` and `Tag` will have 3 requests in total. We probably need a more efficient one by having a generic `Tag()` function by leveraging `FetchReference()` and `PushReference()`.

Reply:
Make sense. 
Like `func (r *Repository) ReTag(ctx context.Context, srcRef, destRef string) error`? Need to figure out a new function name.

Reply:
How about `func Tag(ctx context.Context, repo Repository, srcRef, dstRef string) error` in package `registry`?

Reply:
If it's in package `registry`, it feels like it should instead support `func Tag(ctx context.Context, srcRep, dstRepo Repository, srcRef, dstRef string) error`

Reply:
OK

Reply:
@qweeah has created an issue for this: https://github.com/oras-project/oras-go/issues/120